### DISCUSSION TITLE
Loading Data On Startup

### DISCUSSION THREAD

**Discussion by ZachManno** [serenity-rs/serenity#1651] [Q&A] [answered]
Hello,
I am working on a bot that needs some data to be loaded on startup before the client is set up. For example either call an api or read some data from a CSV file, so when a user sends a message I can use this data for part of the response. How would I pass this data from main() to the handler? 

Thank you!!

**Comment by ZachManno** [comment]
This might answer my question, but looks like the approach in the SQL DB example can be used?

https://github.com/serenity-rs/serenity/blob/current/examples/e16_sqlite_database/src/main.rs#L3

**Comment by vicky5124** [comment, marked as answer]
There's an example explicitly showing this, it's called [global data](https://github.com/serenity-rs/serenity/blob/current/examples/e12_global_data/src/main.rs), and it generates data in main, and passes it to the Client, and then it can be accessed from Context

**Reply by ZachManno** [reply]
Thank you!
